src/sentry/workflow_engine/endpoints/serializers.py

```diff
@@ -361,6 +366,43 @@ def serialize(self, obj: Workflow, attrs: Mapping[str, Any], user, **kwargs) ->
         }
 
 
+@dataclass(frozen=True)
+class WorkflowGroupHistory:
+    group: Group
+    count: int
+    last_triggered: datetime
+    event_id: str
+
+
+class WorkflowFireHistoryResponse(TypedDict):
+    group: BaseGroupSerializerResponse
+    count: int
+    lastTriggered: datetime
+    eventId: str
+
+
+class WorkflowGroupHistorySerializer(Serializer):
+    def get_attrs(
+        self, item_list: Sequence[WorkflowFireHistory], user: Any, **kwargs: Any
+    ) -> MutableMapping[Any, Any]:
+        serialized_groups = {
+            g["id"]: g for g in serialize([item.group for item in item_list], user)
```

Comment:
nit: `g` is a little hard on the readability, but i feel like this is just trying to keep it in 1 line for list comprehension so..... :disappear: